src/policy/write_policy.rs

```diff
@@ -71,10 +71,18 @@ pub struct WritePolicy {
 impl WritePolicy {
     /// Create a new write policy instance with the specified generation and expiration parameters.
     pub fn new(gen: u32, exp: Expiration) -> Self {
-        let mut wp = WritePolicy::default();
-        wp.generation = gen;
-        wp.expiration = exp;
-        wp
+        WritePolicy {
+            base_policy: BasePolicy::default(),
+            record_exists_action: RecordExistsAction::Update,
+            generation_policy: GenerationPolicy::None,
+            commit_level: CommitLevel::CommitAll,
+            generation: gen,
+            expiration: exp,
+            send_key: false,
+            respond_per_each_op: false,
+            durable_delete: false,
+            filter_expression: None,
+        }
```

Comment:
I'm not quite sure I agree with this change. Was this suggested by Clippy? Now you have to maintain the default values for properties like `record_exists_action`, `generation_policy`, etc. in two separate places, rather than having them contained only in `impl Default for WritePolic`.

Reply:
Yes, that was the reason i changed it. I also dont think that this makes much more sense than how it was before.

Reply:
Why not follow the [suggested remediation](https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default) for the `field_reassign_with_default` lint and use the [functional update syntax](https://doc.rust-lang.org/reference/expressions/struct-expr.html#functional-update-syntax), e.g.

```suggestion
        Self {
            generation: gen,
            expiration: exp,
            ..WritePolicy::default()
        }
```

I'm not sure when this syntax was introduced? Maybe this would raise the min. required Rust version even more, if it was only added after Rust 1.42. But looks nice and clean, and is as maintainable as the orig.!

Reply:
I think we can go with rust 1.49 as min compiler version. As far as i know there are no breaking backwards changes.